contracts/multisig-prover/src/msg.rs

```diff
@@ -15,6 +15,11 @@ pub enum ExecuteMsg {
     // Queries the gateway for actual message contents
     #[permission(Any)]
     ConstructProof(Vec<CrossChainId>),
+    #[permission(Any)]
+    ConstructProofWithPayload {
+        message_id: CrossChainId,
+        payload: HexBinary,
+    },
```

Comment:
i think we still want to preserve the ability to batch messages

Reply:
I don't see any benefit in adding batching support here at this time. The Relayer does not support batching for these kinds of messages.
It will also be too complex to implement, especially since the Relayer needs to do 2-3 steps in case of DeployInterchainToken.
InterchainTransfer could be batched, but I don't see how we can do it easily through the Amplifier API at this time.

Reply:
the relayer should support batching for these kinds of messages. not supporting batching can have performance implications. the amplifier api exposes a generic broadcast endpoint that you can use, and you can implement the batching yourself in the relayer. we can address this after launch, but you might as well at least add the batch capability to the contracts now, and then add relayer support when you can

Reply:
Done